src/nodeglpk.hpp

```diff
@@ -167,6 +176,35 @@ class EventEmitterDecorator : public ReentrantCWorker {
      Nan::AsyncWorker* decorated_;
 };
 
+class MemStatsDecorator : public Nan::AsyncWorker {
+ public:
+    MemStatsDecorator(Nan::AsyncWorker* decorated, std::shared_ptr<glp_memstats> memstats)
+        : Nan::AsyncWorker(nullptr), decorated_(decorated), memstats_(memstats) {}
+
+    virtual void HandleOKCallback() override { }
+    virtual void HandleErrorCallback() override { } 
+
+    virtual void WorkComplete() override {
+        MemStatsGuard mguard{memstats_};
+        decorated_->WorkComplete();
+        Nan::AsyncWorker::WorkComplete();
+    }
+
+    virtual void Execute() override {
+        MemStatsGuard mguard{memstats_};
+        decorated_->Execute();
+    }
+
+    virtual void Destroy() override {
+        MemStatsGuard mguard{memstats_};
+        decorated_->Destroy();
+        Nan::AsyncWorker::Destroy();
```

Comment:
ditto to above

Reply:
As above with WorkComplete, but in this case, it's important:

Nan::AsyncWorker#Destroy: 
```C++
  virtual void Destroy() {
      delete this;
  }
```
And that's the only place it happens; So if I don't call Nan::AsyncWorker::Destroy() for each decorator, I'll leak (one could argue that this is why smart-pointers exist, or why "`Destroy()` is somehow more convenient than `delete` but that's a Nan issue)